src/main/java/Model/FEN.java

```diff
@@ -0,0 +1,187 @@
+package Model;
+
+
+import Model.pieces.Piece;
+
+import static java.util.Objects.isNull;
+
+public class FEN {
+
+    private String[] fenPosition = new String[8];
+    private String[] fenBreaks = {"/","/","/","/","/","/","/"," ",};
+    private Boolean whitesTurn = false;
+    private Boolean[] whiteCastling = {true, true};
+    private Boolean[] blackCastling = {true, true};
+    private Piece enPassantPiece = null;
+    private int halfmove = 0;
+    private int fullmove = 0;
+    private String output;
+
+    public FEN(Piece[][] piece2DArray){
+        for (int rank = 0; rank < 8; rank++){
+                updateFENPosition(piece2DArray[rank],rank);
+            }
+    }
+
+    //with all the pieces in the rank, update the appropriate FEN rank in FENPosition
+    public void updateFENPosition(Piece[] pieceRank, int rank){
+        String output = "";
+        int emptySquare = 0;
+        for (Piece piece:pieceRank) {
+            if (isNull(piece)) {
+                emptySquare += 1;
+            } else {
+                if (!(emptySquare == 0)) {
+                    output += emptySquare;
+                    emptySquare = 0;
+                }
+                output += getFENletter(piece);
```

Comment:
if encapsulating the fen inside `PieceType`, this just becomes `output += piece.getPieceType().getFenRepresentation(piece.getIsWhite())`.

Why not go even further and make `piece.getFenRepresentation()` a method - it would then have all the info it needs internally.

Reply:
Implemented.